contributions/authority/auth0/src/main/java/com/yahoo/athenz/auth/oauth/auth0/Auth0Jwt.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.auth.oauth.auth0;
+
+import java.util.LinkedHashMap;
+import com.yahoo.athenz.auth.AuthorityConsts;
+import com.yahoo.athenz.auth.oauth.token.DefaultOAuthJwtAccessToken;
+import com.yahoo.athenz.auth.oauth.token.OAuthJwtAccessToken;
+import com.yahoo.athenz.auth.util.AthenzUtils;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.RequiredTypeException;
+
+/**
+ * Custom for Auth0 JWT access token format
+ */
+public class Auth0Jwt extends DefaultOAuthJwtAccessToken {
+
+    private static String claimClientId = "https://myapp.example.com/client_id";
```

Comment:
where/how are we using this example.com/client_id? do we need a such value as default or is the caller expected to always specify a value?

Reply:
Auth0 has restriction on custom field format, so we need the "https://myapp.example.com/" part as the namespace.
https://auth0.com/docs/api-auth/tutorials/adoption/scope-custom-claims#custom-claims

Also, as we enforced the client ID and cnf checking in the new authority class, the claim name should be specified. For the default value, I just use the value can be found in auth0's docs.